benchopt/base.py

```diff
@@ -103,59 +102,42 @@ def is_installed(cls, env_name=None):
 
     @classmethod
     def install(cls, env_name=None, force=False):
-        """Install the class in the given virtual env.
+        """Install the class in the given conda env.
 
         Parameters
         ----------
         env_name: str or None
-            Name of the environment where the class should be installed. If
+            Name of the conda env where the class should be installed. If
             None, tries to install it in the current environment.
         force : boolean (default: False)
-            If set to True, first tries to uninstall the class from the
-            environment before installing it.
+            If set to True, forces reinstallation when using conda.
 
         Returns
         -------
         is_installed: bool
             True if the class is correctly installed in the environment.
         """
-        # uninstall the class that requires a force reinstall
-        if force:
-            cls.uninstall(env_name=env_name)
+        print(f"Installing {cls.name} in {env_name}:...",
+              end='', flush=True)
+        try:
+            if cls.install_cmd == 'conda':
+                conda_install_in_env(*cls.requirements_install,
+                                     env_name=env_name, force=force)
+            elif cls.install_cmd == 'bash':
+                bash_install_in_env(cls.install_script, env_name=env_name)
```

Comment:
is it really bash you need here? or just a shell? any chance windows can supported for at least python solvers?

Reply:
We just need a shell but most script here will require compilers I guess so not sure if there is much hope to make this work for windows. That is why using conda for all would be a nice solution.

Reply:
As it is probably just shell, I renamed this option `shell`.

Note that I also install `conda` compilers so maybe ther is a way to compile with conda cross platform but not sure how this would work (we should maybe ask jeremie about this).